### DISCUSSION TITLE
Business Rules that are inside an object

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3569] [Questions] [answered]
We have a few objects with business rules defined inside these objects and no other objects can call these 'inside' rules (they are private).
Do we still need to use RuleUri.AddQueryParameter(......, ....)?
We are accessing properties of the object inside these rules with ReadProperty and LoadProperty methods.

Kind Regards

**Comment by rockfordlhotka** [comment, marked as answer]
iirc, the RuleUri is only really used when building a list of broken rules, and exists so that the consumer of that list of broken rules has some unique way of identifying each broken rule.

In other words, the list could contain a number of broken "required" rules, and the only way to know which broken rule applies to which property(ies) is via the RuleUri.

If you aren't using any of the features where you get a list of broken rules for an object or object graph, then the RuleUri (I think) isn't used at all.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka I didn't use RuleUri.AddQueryParameter(......, .... ) so hopefully nothing drastic will happen :)
